Saucelabs branch

scottynomad opened the pull request:
This adds some config directives and a nose argument to enable running selenium tests backed by saucelabs.

There are also a couple of test fixes and some relaxing of timeouts to allow for the network link.

Requires running sauce connect (tunneling proxy) as:

java -jar /idealist/Sauce-Connect.jar sbuser sbapikey

invocations of the alfajor tests were looking like:

bin/nosetests --alfajor-config=alfajor.saucelabs-real.ini --browser=firefox --saucelabs="browser-version=7" --saucelabs="os=Windows 2003" -w tests tests.browser.test_browser:test_wait_for_duration --pdb-f --pdb

There are a handful of failing browser tests on firefox (multiple cookie ones) and a slightly larger handful not yet working on ie (the ajax and js wait for conditions).

scottynomad committed:
Updated apiclient to work with Werkzeug >= 0.8

scottynomad committed:
Changed the test webapp port to 8888 which is supported by the saucebox proxy (sauce connect).

scottynomad committed:
Switched up the simple browser test to work with the strange (de)normalization of html coming back from IE.

scottynomad committed:
Increase the duration wait_for value so it's less likely to fail on slow network links.

scottynomad committed:
Relax the timeout on ajax test and handle the value error differently to get more info from it.

scottynomad committed:
Pass the selenium driver a little extra timeout so our wait_for=duration tests don't fail over network links.

scottynomad committed:
Tweaks to alfajor to allow running tests against saucelabs with their sauce connect proxy tool.

dcolish commented on alfajor/_management.py:
```diff
@@ -181,8 +181,17 @@ class _ManagerFactory(object):
 
         """
         # --alfajor-config overrides any config data in code
+        # if an unqualified name is passed then search in the declared path
```
We used to require q qualified name to avoid duplicating the search code found a few lines below this diff. It would be good to avoid the duplication

dcolish commented on alfajor/browsers/managers.py:
```diff
@@ -54,6 +55,23 @@ class SeleniumManager(object):
+    def _browser_cmd(self):
+        browser_cmd = {}
+        for key, value in self.config.items():
+            if key.startswith('saucelabs-'):
```
This makes me think there should be a distinct saucelabs section in the config rather than doing this dance.

scottynomad committed:
- Smoothed out the saucelabs integration by treating it as a separate backend
derived from selenium.

- Fixed some more tests

- Addressed Dan's comments including removal of the extra config search code.

Multiple cookie tests are still failing for saucelab, think their proxy
is mangling them.

scottynomad commented:
After grok'ing that I can specify custom browser implementations in config I think most or all of this code could be removed or shifted into a contrib / example module.  The only thing that would be lost would be the nose cmd line flag integration which is not strictly necessary.
